### ISSUE TITLE
Improper `@example` parsing when an at-sign `@` is used in the code-block.

### ISSUE THREAD

**Issue by JHawkley** [microsoft/TypeScript#38922]
Noticed an issue with the `@example` JSDoc tag when the example happens to use an `@` sign, such as for an email address (my case).

- VSCode Version: 1.45.1
- OS Version: Ubuntu 20.04

Steps to Reproduce:

1. Make an `@example` JSDoc tag similar to the below:
```js
/**
 * @example Example Output
 * {
 *   name: "Testy McTest",
 *   email: "testy.mctest@not_an_email.com"
 *   gender: "ambiguous"
 * }
 */
const theProfile = getProfile();
```
2. Observe the tooltip for `theProfile`.  VSCode parses the `@not_an_email` as a JSDoc tag, breaking the code formatting.  It will do this even if the example code is fenced in back-ticks, with slightly different output.

Does this issue occur when all extensions are disabled?: N/A

**Comment by mjbvz** [comment]
Potential fixes:

- Require the `@` always be prefixed by a whitespace, `{`, or the start of the line (this would not fix other cases such as `email: "a @ b"`

- Only parse inline jsdoc tags if they are known, such as `@link`

**Comment by MartinJohns** [comment]
Duplicate of #35310.

**Comment by sandersn** [comment]
Duplicate of #39371, and fixed in #42364.
